src/h5.cpp

```diff
@@ -181,18 +180,18 @@ append_ret h5d::append_data_1d(T const *data, hsize_t nlen) {
       sext.at(i) = sext2.at(i);
     }
     auto t3 = CLK::now();
-    LOG(Sev::Debug, "h5d::append_data_1d set_extent: {} + {}",
-        std::chrono::duration_cast<MS>(t2 - t1).count(),
-        std::chrono::duration_cast<MS>(t3 - t2).count());
+    Logger->trace("h5d::append_data_1d set_extent: {} + {}",
+                  std::chrono::duration_cast<MS>(t2 - t1).count(),
+                  std::chrono::duration_cast<MS>(t3 - t2).count());
   }
 
-  if (log_level >= 9) {
-    LOG(Sev::Debug, "try to get the dsp dims:");
+  if (Logger->should_log(spdlog::level::trace)) {
```

Comment:
```suggestion
```

Reply:
I think this `if` should be there, it is preventing getting dimensions and executing a loop if the log level is set above `trace`, which would usually be the case.

Reply:
I think that this is a false optimisation. Better to remove useless code.

Reply:
@matthew-d-jones By the way, committing those changes was actually a mistake by me. So if you have a strong opinion about this, I can revert it.